proposals/0022-multi-stake.md

```diff
@@ -0,0 +1,178 @@
+more, and now your stake account has two delegations: the initial one, and the
+newly activating one.
+
+Once the new `Delegation` is active, a permissionless `update` instruction follows
+the existing `merge` logic to consolidate the two `Delegation`s into one.
+
+The same roughly applies for removing a small stake. You `deactivate` a portion
+of it, which adds another delegation to your stake account: the initial one, and
+the newly deactivating one.
+
+### Redelegate
+
+With another `Delegation` instance, you can redelegate from one validator to another
+within the same account.
+
+It works the same as the existing "redelegate" instruction, except the lamports
+all stay in the same account, and the second `Delegation` covers the redelegation.
+
+The permissionless `update` instruction clears out the first one once it's inactive.
+
+### Single validator stake pool
+
+A single-validator stake pool program can manage small stakes with 100% efficiency
+using "multi stakes".
+
+Since there's a total of *three* `Delegation` instances, small stakes can be added
+and removed at the same time. One instance covers the main amount, another covers
+new activations, and the last one covers deactivations.
+
+For small stake deposit, you simply transfer the lamports and activate the new amount.
+
+For small stake withdrawal, the program deactivates the small stake and provides
+a claimable ticket to the user, used to claim their lamports after deactivation.
+
+### Runtime
+
+The main difference to the runtime is processing up to three `Delegation`s
+per account.
+
+There's minor impact on the stakes cache, which gets a little bigger, but the
+maximum is bounded by the minimum delegation amount.
+
+There's minor impact on rewards payout, since the time spent calculating the rewards
+amount is negligible compared to the time spent storing the accounts. The number
+of account storages stays the same.
+
+## Impact
+
+Validators likely see a small increase in memory usage from additional `Delegation`
+entries in the stakes cache, and perhaps a tiny increase in rewards payout time,
+but nothing substantial.
+
+Dapp developers have to deal with a new stake account type. This may break programs
+that try to deserialize any stake account.
+
+The network benefits as a whole:
+
+* stake accounts are more flexible for adding or removing stake, without splits and merges
+* stakers can easily redelegate without losing funds
+* single-validator stake pools can be just about 100% capital efficient
+
+## Security Considerations
+
+It's risky to change the runtime, especially the stake program and rewards payout.
+Thankfully, since this all operates on the level of the `Delegation`, as long
+as we pass all the instances as needed, none of the most sensitive parts need
+to change.
+
+## Drawbacks *(Optional)*
+
+We could just change the stakes cache include small stakes, but not in rewards
+payout, and everything mostly works, but it's ultimately a temporary solution.
```

Comment:
this was already covered sufficiently in `Alternatives Considered`?

afaik this section is intended to elucidate drawbacks with the solution as proposed.  there's kinda some of that in `Backward Compatibility` below.